lib/portlayer/event/event_manager.go

```diff
@@ -15,38 +15,31 @@
 package event
 
 import (
-	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/lib/portlayer/event/collector"
+	"github.com/vmware/vic/lib/portlayer/event/events"
 )
 
+// EventManager will provide a basic event pub/sub implementation
 type EventManager interface {
 
-	// AddMonitoredObject will add the object for event listening
-	AddMonitoredObject(ref string) error
+	// RegisterCollector a collector with the manager
+	RegisterCollector(collector.Collector)
 
-	// RemoveMonitoredObject will remove the object from event listening
-	RemoveMonitoredObject(ref string)
+	// Collectors returns registered collectors
+	Collectors() map[string]collector.Collector
 
-	// Register will for event callbacks
-	Register(caller string, callback func(Event, *session.Session))
+	// Subscribe for event callbacks
+	Subscribe(event interface{}, caller string, callback func(events.Event))
```

Comment:
I thought about that -- but was concerned with the definition and maintenance of a topic.  I wasn't happy about the reflection overhead, but since it's used minimally I thought it to be acceptable.

If you think it's worth it I can move to a string topic -- I'll define the topics as a constant in the event package.  Then as new events are created, removed, etc...the constant and the identification method (this string == this event, etc) will need to be updated.  Is that what you're thinking?